### PR TITLE
Placeholder input validation

### PR DESCRIPTION
**Pull request by xlcheng1**
### Motivation
- If customers accidentally use a placeholder value (or value that cannot be expanded by wordexp) for certain parameters in config file, device client will have a segmentation fault because of ExtractExpandedPath() function.

#### Change summary
- Added checks when parsing config file
- Added a placeholderInput test 

#### Revision diff summary
If there is more than one revision, please explain what has been changed since the last revision.

### Testing
 **Is your change tested? If not, please justify the reason.**  
 **Please list your testing steps and test results.** 
- CI test run result: <link>

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/config/Config.cpp [hunk 1]
```diff
@@ -1239,24 +1216,6 @@ bool PlainConfig::PubSub::Validate() const
     {
         return true;
     }
-    if (!publishTopic.has_value() || publishTopic->empty())
```
Comment (BolongZhang-AWS) [inline source/config/Config.cpp:1242]: Also why is this part removed?

File: source/config/Config.cpp [hunk 2]
```diff
@@ -354,10 +354,6 @@ bool PlainConfig::Validate() const
         LOGM_ERROR(Config::TAG, "*** %s: Certificate is missing ***", DeviceClient::DC_FATAL_ERROR);
         return false;
     }
-    else if (!FileUtils::IsValidFilePath(cert->c_str()))
```
Comment (RogerZhongAWS) [inline source/config/Config.cpp:357]: I am a little bit confused on this, don't we still need to have this if we pass in params via CLI?

File: source/config/Config.cpp [hunk 3]
```diff
@@ -1047,22 +1040,6 @@ bool PlainConfig::FleetProvisioning::Validate() const
         return false;
     }
 
-    if (csrFile.has_value() && !csrFile->empty())
-    {
-        if (!FileUtils::IsValidFilePath(csrFile->c_str()))
```
Comment (RogerZhongAWS) [inline source/config/Config.cpp:1052]: here as well

**Comment by xlcheng1** [comment]
closing this pr because PR #288 is providing better approach solving the issue.
